application/helpers/expressions/em_manager_helper.php

```diff
@@ -5055,15 +5055,21 @@ private function _UpdateValuesInDatabase($finished = false)
             SurveyDynamic::sid($this->sid);
             $oSurvey = new SurveyDynamic();
 
-            $iNewID = $oSurvey->insertRecords($sdata);
-            if ($iNewID) {    // Checked
+            try {
+                $iNewID = $oSurvey->insertRecords($sdata);
+                if (!$iNewID) {
+                    throw new Exception("Error, no entry id was returned.", 1);
+                }
                 $srid = $iNewID;
                 $_SESSION[$this->sessid]['srid'] = $iNewID;
-            } else {
+            } catch (Exception $e) {
                 $srid = null;
                 $message .= $this->gT("Unable to insert record into survey table"); // TODO - add SQL error?
-                submitfailed($this->gT("Unable to insert record into survey table"));
+                $query = $e->getMessage();
+                $trace = $e->getTraceAsString();
+                submitfailed($this->gT("Unable to insert record into survey table"), $query . "\n\n" . $trace);
```

Comment:
I'm having a silent fail here. Isn't `submitfailed` suppose to end survey taking?

Reply:
> La propietat "SurveyDynamic"."datestamp" no es troba definida.

Reply:
In other places in EM, it's used as

```php
$message = submitfailed($this->gT('The data could not be saved because the response does not exist in the database.'));
LimeExpressionManager::addFrontendFlashMessage('error', $message, $this->sid);
return;
```

Reply:
https://bugs.limesurvey.org/view.php?id=19335

Reply:
> I'm having a silent fail here. Isn't `submitfailed` suppose to end survey taking?

Were you getting an email message at least?

Reply:
I didn't check the email. Customer didn't mention it. But yeah, they should have.